test/gtest/common/mem_buffer.cc

```diff
@@ -77,11 +77,25 @@ bool mem_buffer::is_rocm_managed_supported()
 {
 #if HAVE_ROCM
     int device_id, has_managed_mem;
-    return ((hipGetDevice(&device_id) == hipSuccess) &&
+    hipError_t ret;
+    void * dptr;
+    hipPointerAttribute_t attr;
+    if ((hipGetDevice(&device_id) == hipSuccess) &&
             (hipDeviceGetAttribute(&has_managed_mem,
                                    hipDeviceAttributeManagedMemory,
                                    device_id) == hipSuccess) &&
-            has_managed_mem);
+                                   has_managed_mem) {
+        ret = hipMallocManaged(&dptr, 64);
```

Comment:
1. IMO would be nicer to have series of checks and return false for any failed check, instead of nested 'if' s
2. pls check clang-format

Reply:
Ok i should have addressed those. I am assuming that the valgrind errors in the CI are not due to my pr